pkg/networking/middleware/response.go

```diff
@@ -0,0 +1,123 @@
+package middleware
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/snyk/error-catalog-golang-public/snyk"
+	"github.com/snyk/error-catalog-golang-public/snyk_errors"
+)
+
+type ResponseMiddleware struct {
+	next       http.RoundTripper
+	errHandler func(error, context.Context) error
+}
+
+func NewReponseMiddleware(roundTriper http.RoundTripper, errHandler func(error, context.Context) error) *ResponseMiddleware {
+	return &ResponseMiddleware{
+		next:       roundTriper,
+		errHandler: errHandler,
+	}
+}
+
+func errFromResponse(res *http.Response) error {
+	if res.StatusCode >= 200 && res.StatusCode < 300 {
```

Comment:
Suggestion: use constants in http package

Reply:
Actually you might be able to remove this logic at all, if you catch this cases in the code below.